### PR TITLE
Add minKubeVersion validation to templates

### PR DESCRIPTION
**Pull request by dinhxuanvu**
Require minKubeVersion to be SemVer formated (Major.Minor.Patch).

Signed-off-by: Vu Dinh <vdinh@redhat.com>

### CODE REVIEW

File: deploy/ocp/manifests/0.8.1/0000_50_olm_02-clusterserviceversion.crd.yaml [hunk 1]
```diff
@@ -65,6 +65,7 @@ spec:
             minKubeVersion:
               type: string
               description: Minimum kubernetes version requirement on the server to deploy operator
+              pattern: ^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(-(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(\.(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*)?(\+[0-9a-zA-Z-]+(\.[0-9a-zA-Z-]+)*)?$
```
Comment (njhale) [inline deploy/ocp/manifests/0.8.1/0000_50_olm_02-clusterserviceversion.crd.yaml]: Anything starting with a 'v' will not be matched by this regular expression.

File: manifests/0000_50_olm_09-operatorgroup.crd.yaml [hunk 2]
```diff
@@ -1,3 +1,5 @@
+---
+# Source: olm/templates/0000_50_olm_09-operatorgroup.crd.yaml
```
Comment (njhale) [inline manifests/0000_50_olm_09-operatorgroup.crd.yaml]: These need to be regenerated. Some recipes in our `Makefile` rely on the GNU flavors of certain utilities for this. Looks like you are using the BSD versions. We should really containerize `make release`.

File: deploy/upstream/values.yaml [hunk 3]
```diff
@@ -7,21 +7,21 @@ writeStatusName: '""'
 olm:
   replicaCount: 1
   image:
-    ref: quay.io/openshift/origin-operator-lifecycle-manager:latest
+    ref: quay.io/coreos/olm@sha256:995a181839f301585a0e115c083619b6d73812c58a8444d7b13b8e407010325f
```
Comment (njhale) [inline deploy/upstream/values.yaml]: You can just leave these changes out for now.

**Comment by dinhxuanvu** [comment]
@njhale Would you mind taking a look?

**Comment by njhale** [comment]
Looks like you're hitting an issue with the new `anyOf` validation. I opened #744 to fix the problem.

**Comment by njhale** [comment]
Sweet, this looks great!
/lgtm

**Pull request merged**